### ISSUE TITLE
DHT sniffer is no longer working

### ISSUE THREAD

**Issue by cshuaimin** [ipfs-search/ipfs-search#62]
In the recent go-ipfs update:
-  `.event` will never equal `handleAddProvider` in `ipfs log tail`
- but `handleAddProvider` is available in Opentracing's logs(maybe)
- but  Opentracing's log no longer outputs multihash. It only prints the peer ID:
```sh
$ ipfs log tail|grep handleAddProvider
{"Operation":"handleAddProvider","Fields":[{"Key":"peer","Value":"QmYxoZmhx5kiAd8MK5ZgMF2Co2GpH2xXWSm8T6JPnrwzUt"}], <others..>}
```

So code in [dht-snifloop.sh](https://github.com/ipfs-search/ipfs-search/blob/master/provisioning/roles/snifloop/files/bin/dht-snifloop.sh) `ipfs log tail | jq -r 'if .event == "handleAddProvider" then .key else empty end'` is no longer working.

**Comment by dokterbob** [comment]
Do you have any suggestions on how we might tackle this? 
-- 
Verstuurd vanaf mijn Android apparaat met K-9 Mail. Excuseer mijn beknoptheid.

**Comment by cshuaimin** [comment]
I see that if you set the log level of dht to info, the daemon will output the required information:
```sh
$ ipfs daemon &                          
$ ipfs log level dht info
 
20:24:39.052  INFO        dht: received provider <peer.ID TvhUhw> for QmapMVrA8cAhxw8uBd1kuUfx4zGQpi6ed3fhdB9KBjkMP4 (addrs: [/ip4/127.0.0.1/tcp/4001 /ip4/199.247.16.66/tcp/4001 /ip6/::1/tcp/4001 /ip6/2001:19f0:6c01:1d43:5400:1ff:fe7c:bbd4/tcp/4001 /ip4/199.247.16.66/tcp/4001 /ip6/2001:19f0:6c01:1d43:5400:1ff:fe7c:bbd4/tcp/4001]) handlers.go:395
```

So we can start the daemon through the crawler and capture the stdout to sniff new multihashes. But I think maybe there is a better way.

**Comment by cshuaimin** [comment]
Should be fixed in libp2p/go-libp2p-kad-dht#190

**Comment by dokterbob** [comment]
This is not yet fixed as of 0.4.17

**Comment by dokterbob** [comment]
Let's see if we can get this as part of ipfs 0.4.18. For now, I'll just keep things at the latest working version 0.4.15.

**Comment by cshuaimin** [comment]
Ok. I did not notice the version of ipfs you are using before.

**Comment by dokterbob** [comment]
Fixed in go-ipfs 0.4.18rc1, finally allowing us to upgrade (but we'll wait for release just to be sure).

Relevant code: `./ipfs log tail | jq -r 'if .Operation == "handleAddProvider" then .Tags.key else empty end'`

**Comment by dokterbob** [comment]
Closed with #68
